packages/web-features-figuring/src/baseline/index.ts

```diff
@@ -0,0 +1,241 @@
+import assert from "node:assert";
+
+import { Temporal } from "@js-temporal/polyfill";
+
+import { Browser } from "../browser-compat-data/browser";
+import { Compat, defaultCompat } from "../browser-compat-data/compat";
+import { feature } from "../browser-compat-data/feature";
+import { Release } from "../browser-compat-data/release";
+import { browsers } from "./core-browser-set";
+import { isFuture, toDateString, toHighDate } from "./date-utils";
+import { support } from "./support";
+
+interface FeatureSelector {
+  compatKeys: [string, ...string[]];
+  checkAncestors: boolean;
+}
+
+type BaselineStatus = "low" | "high" | false;
+type BaselineDate = string | null;
+
+interface SupportStatus {
+  compatKey?: string;
+  baseline: BaselineStatus;
+  baseline_low_date: BaselineDate;
+  baseline_high_date: BaselineDate;
+  support: Map<Browser, Release | undefined>;
+  toJSON: () => string;
+}
+
+/**
+ * Given a set of compat keys, compute the aggregate Baseline support ("high",
+ * "low" or false, dates, and releases) for those keys.
+ */
+export function computeBaseline(
+  featureSelector: FeatureSelector,
+  compat: Compat = defaultCompat,
+): SupportStatus {
+  const { compatKeys } = featureSelector;
+  const keys = featureSelector.checkAncestors
+    ? compatKeys.flatMap((key) => withAncestors(key, compat))
+    : compatKeys;
+
+  const statuses = keys.map((key) => calculate(key, compat));
+  const support = collateSupport(statuses.map((status) => status.support));
+
+  const keystoneDate = findKeystoneDate(
+    statuses.flatMap((s) => [...s.support.values()]),
+  );
+  const { baseline, baseline_low_date, baseline_high_date } =
+    keystoneDateToStatus(keystoneDate);
+
+  return {
+    baseline,
+    baseline_low_date,
+    baseline_high_date,
+    support,
+    toJSON: function () {
+      return jsonify(this);
+    },
+  };
+}
+
+/**
+ * Compute the Baseline support ("high", "low" or false, dates, and releases)
+ * for a single compat key.
+ */
+function calculate(compatKey: string, compat: Compat): SupportStatus {
+  const f = feature(compatKey);
+  const s = support(f, browsers(compat));
+  const keystoneDate = findKeystoneDate([...s.values()]);
+
+  const { baseline, baseline_low_date, baseline_high_date } =
+    keystoneDateToStatus(keystoneDate);
```

Comment:
Reusing the same logic as above, to find the keystone date and status for a single compat key.